src/main/drivers/pwm_output.c

```diff
@@ -521,7 +522,7 @@ void pwmWriteDshotCommand(uint8_t index, uint8_t motorCount, uint8_t command, bo
             delayMicroseconds(DSHOT_COMMAND_DELAY_US);
 
 #ifdef USE_DSHOT_TELEMETRY
-            pwmStartDshotMotorUpdate(motorCount);
+            while (!pwmStartDshotMotorUpdate(motorCount));
```

Comment:
Hmmmm... blocking seems dangerous.

Reply:
It blocks at most for 7us plus the dshot telemetry frame length. Logically it isn't needed but we could add another timeout here if you think that would be safer for code maintenance in the future when the pwmStartDshotMotorUpdate implementation might change.

Reply:
Ok I added a 1ms timeout.